### PR TITLE
Collapsible JSON component

### PR DESCRIPTION
**Pull request by matv-stripe**
The JSON view in the sandbox would be easier to understand if 

- it was collapsible
- had a scroll container
- let you highlight parts of the source text and see where in the JSON view that source text comes from

This attempts the first part, but a WIP since I have not fixed all the styling.

### CODE REVIEW

File: components/ObjectDisplay.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            <pre style={{ color: valueColor }}>
+              {typeof value === 'string' ? `"${value}"` : String(value)}
+            </pre>
+          )}
+        <pre>{','}</pre>
+      </KeyValueRow>
+    );
+  });
+
+  return (
```
Comment (matv-stripe) [inline components/ObjectDisplay.js:179]: Thanks @ianjabour-stripe for the inspiration here!

But maybe a better solution here is use code mirror folding? - https://codemirror.net/demo/folding.html

**Comment by matv-stripe** [comment]
cc @nkohari-stripe 

r? @mfix-stripe , WDYT of this approach?

File: components/Sandbox.js [hunk 2]
```diff
@@ -381,15 +382,16 @@ export function Sandbox({ height, options }) {
               }}
             />
           )}
-          {mode === 'ast' && (
+          {mode === 'ast' && <ObjectDisplay data={ast} />}
+          {/* {mode === 'ast' && (
```
Comment (mfix-stripe) [inline components/Sandbox.js:386]: Can we just delete this commented out code?

File: components/Sandbox.js [hunk 3]
```diff
@@ -381,15 +382,16 @@ export function Sandbox({ height, options }) {
               }}
             />
           )}
-          {mode === 'ast' && (
+          {mode === 'ast' && <ObjectDisplay data={ast} />}
```
Comment (mfix-stripe) [inline components/Sandbox.js:385]: Let's also use this for the `transform` mode directly above

File: components/Sandbox.js [hunk 4]
```diff
@@ -66,15 +67,15 @@ export function useMarkdocCode(code) {
...
+  const content = React.useMemo(() => Markdoc.transform(ast, config), [
+    ast,
+    config
+  ]);
 
-  const errors = React.useMemo(
-    () => Markdoc.validate(ast, config),
-    [ast, config]
-  );
+  const errors = React.useMemo(() => Markdoc.validate(ast, config), [
```
Comment (mfix-stripe) [inline components/Sandbox.js:75]: Why is your prettier different? ahah

File: components/ObjectDisplay.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    <div style={{ background: 'var(--black-medium)' }}>
+      <LineStack
+        rootKey={rootKey}
+        hasTrailingComma={!isRoot}
+        onClick={() => setCollapsed(true)}
+        isArray={Array.isArray(data)}
```
Comment (mfix-stripe) [inline components/ObjectDisplay.js:185]: Feels like you don't need this prop—can just be derived from `data`

File: components/ObjectDisplay.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                style={{ color: 'rgb(163, 172, 185)', lineHeight: 0 }}
+              >{` ... `}</span>
+              <span
+                style={{ lineHeight: 0, color: 'rgb(163, 172, 185)' }}
+              >{`${childCount} item${childCount === 1 ? '' : 's'} `}</span>
+              {trailing}
+            </>
+          )}
+        </pre>
+        {!isRoot && <pre>{','}</pre>}
```
Comment (mfix-stripe) [inline components/ObjectDisplay.js:142]: Do we need this `isRoot` thing?

**Review by mfix-stripe — CHANGES_REQUESTED** [review]
PTAL @matv-stripe

**Comment by mfix-stripe** [comment]
PTAL @matv-stripe!

**Comment by matv-stripe** [comment]
I like it, thanks for fixing this up!

I think it might be cleaner if it defaulted to something like this instead of showing everything initially:
image

Maybe code folding might be an easier solution - but I am good with merging this because it is an improvement!

**Comment by matv-stripe** [comment]
ptal @mfix-stripe , I cant approve this because its my branch 😆

**Review by mfix-stripe — APPROVED** [review]

**Comment by mfix-stripe** [comment]
@matv-stripe thanks! How does this look: 
image

**Comment by matv-stripe** [comment]
Its great! @mfix-stripe

**Pull request merged**